api/kessel/inventory/v1beta1/k8s_cluster_detail.proto

```diff
@@ -16,60 +14,64 @@ message K8sClusterDetail {
   // the aggregate status of the cluster
   enum ClusterStatus {
     CLUSTER_STATUS_UNSPECIFIED = 0;
-    CLUSTER_STATUS_READY = 1;
-    CLUSTER_STATUS_FAILED = 2;
-    CLUSTER_STATUS_OFFLINE = 3;
+    CLUSTER_STATUS_OTHER = 1; 
+    READY = 2;
+    FAILED = 3;
+    OFFLINE = 4;
   }
 
   // The kubernetes vendor
   enum KubeVendor {
     KUBE_VENDOR_UNSPECIFIED = 0;
     KUBE_VENDOR_OTHER = 1;
-    KUBE_VENDOR_AKS = 2;
-    KUBE_VENDOR_EKS = 3;
-    KUBE_VENDOR_IKS = 4;
-    KUBE_VENDOR_OPENSHIFT = 5;
-    KUBE_VENDOR_GKE = 6;
+    AKS = 2;
+    EKS = 3;
+    IKS = 4;
+    OPENSHIFT = 5;
+    GKE = 6;
   }
 
   // The platform on which this cluster is hosted
   enum CloudPlatform {
     CLOUD_PLATFORM_UNSPECIFIED = 0;
-    CLOUD_PLATFORM_NONE_UPI = 1;
-    CLOUD_PLATFORM_BAREMETAL_IPI = 2;
-    CLOUD_PLATFORM_BAREMETAL_UPI = 3;
-    CLOUD_PLATFORM_AWS_IPI = 4;
-    CLOUD_PLATFORM_AWS_UPI = 5;
-    CLOUD_PLATFORM_AZURE_IPI = 6;
-    CLOUD_PLATFORM_AZURE_UPI = 7;
-    CLOUD_PLATFORM_IBMCLOUD_IPI = 8;
-    CLOUD_PLATFORM_IBMCLOUD_UPI = 9;
-    CLOUD_PLATFORM_KUBEVIRT_IPI = 10;
-    CLOUD_PLATFORM_OPENSTACK_IPI = 11;
-    CLOUD_PLATFORM_OPENSTACK_UPI = 12;
-    CLOUD_PLATFORM_GCP_IPI = 13;
-    CLOUD_PLATFORM_GCP_UPI = 14;
-    CLOUD_PLATFORM_NUTANIX_IPI = 15;
-    CLOUD_PLATFORM_NUTANIX_UPI = 16;
-    CLOUD_PLATFORM_VSPHERE_IPI = 17;
-    CLOUD_PLATFORM_VSPHERE_UPI = 18;
-    CLOUD_PLATFORM_OVIRT_IPI = 19;
+    CLOUD_PLATFORM_OTHER = 1;
+    NONE_UPI = 2;
+    BAREMETAL_IPI = 3;
+    BAREMETAL_UPI = 4;
+    AWS_IPI = 5;
+    AWS_UPI = 6;
+    AZURE_IPI = 7;
+    AZURE_UPI = 8;
+    IBMCLOUD_IPI = 9;
+    IBMCLOUD_UPI = 10;
+    KUBEVIRT_IPI = 11;
+    OPENSTACK_IPI = 12;
+    OPENSTACK_UPI = 13;
+    GCP_IPI = 14;
+    GCP_UPI = 15;
+    NUTANIX_IPI = 16;
+    NUTANIX_UPI = 17;
+    VSPHERE_IPI = 18;
+    VSPHERE_UPI = 19;
+    OVIRT_IPI = 20;
+   
+    
   }
 
   // The OCP cluster ID or ARN etc for *KS
-  string external_cluster_id = 219571597 [ (validate.rules).string.min_len = 1 ];
+  string external_cluster_id = 219571597;
```

Comment:
These validate.rules guard against having an empty `external_cluster_id` - without this, `external_cluster_id` won't be required.

Is this OK?

Reply:
yes